File: byoc-samples/byoc-managed-identity-keyvault/byoc-managed-identity-keyvault-python/identity.py
from azure.identity import ManagedIdentityCredential
from azure.keyvault.secrets import SecretClient
import indentityconfig as cfg

credential = ManagedIdentityCredential()
keyVaultUrl = cfg.key_vault["vault_url"]
print("About to connect to: ", keyVaultUrl)
client = SecretClient(keyVaultUrl, credential, logging_enable=True)

def getSecret(name):
    try:
        print("About to get secret: " + name)
        secret = client.get_secret(name)
        return "Successfully got the value of secret {} from Key Vault {}: {}".format(name, keyVaultUrl, secret.value)
    except Exception as e:
        print(e)
        return "Failed to get secret {} from Key Vault {} due to {}".format(name, keyVaultUrl, e.__str__())

def setSecret(name, value):
    try:
        print("About to set secret: ({}, {})".format(name, value))
        secret = client.set_secret(name, value)
        return "Successfully got the value of secret {} from Key Vault {}: {}".format(name, keyVaultUrl, secret.value)
    except Exception as e:
        print(e)
        return "Failed to set secret {} in Key Vault {} due to {}".format(name, keyVaultUrl, e.__str__())

File: byoc-samples/byoc-managed-identity-keyvault/byoc-managed-identity-keyvault-node/identity.js
const { SecretClient } = require("@azure/keyvault-secrets");
const { ManagedIdentityCredential } = require("@azure/identity");
const util = require('util');

const credential = new ManagedIdentityCredential()

const url = process.env.AZURE_KEYVAULT_URI;
console.log("About to connect to: ", url);
const client = new SecretClient(url, credential);

async function getSecret(name) {
    console.log("About to get secret: " + name);
    try {
        var secret = await client.getSecret(name);
        return util.format("Successfully got the value of secret %s from Key Vault %s: %s",
                    name, url, secret.value);
    } catch (err) {
        console.log(err)
        return util.format("Failed to get secret %s from Key Vault %s due to %s", name,
                    url, err.message);
    }
}

async function setSecret(name, value) {
    console.log(util.format("About to set secret: (%s, %s)"), name, value);

    try {
        await client.setSecret(name, value);
        return util.format("Successfully set secret %s in Key Vault %s", name, url);
    } catch (err) {
        console.log(err)
        return util.format("Failed to set secret %s in Key Vault %s due to %s", name,
                    url, err.message);
    }
}

module.exports = {getSecret, setSecret};